src/pytypes/onefuzztypes/_monkeypatch.py

```diff
@@ -0,0 +1,50 @@
+# TODO: Remove once `smart_union` like support is added to Pydantic
+#
+# Written by @PrettyWood
+# Code from https://github.com/samuelcolvin/pydantic/pull/2092
+#
+# Original project licensed under the MIT License.
+
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple, Union
+
+from pydantic.fields import ModelField
+from pydantic.typing import get_origin
+
+if TYPE_CHECKING:
+    from pydantic.error_wrappers import ErrorList
+    from pydantic.types import ModelOrDc
+
+    ValidateReturn = Tuple[Optional[Any], Optional[ErrorList]]
+    LocStr = Union[Tuple[Union[int, str], ...], str]
+
+orig = ModelField._validate_singleton
```

Comment:
```suggestion

orig = ModelField._validate_singleton
```

Reply:
It might improve at-a-glance readability if we renamed these along the lines of:
- `orig` -> `upstream_validate_singleton`
- `wrapper` -> `patched_validate_singleton`